tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,12 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (NonEmptySet (xs:Set<int>)) =
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
This doesn't look correct to me.

As an example, couldn't `xs` conceivably be `set [0]`? If so, all shuffles are the same as the original input list:

```F#
> let xs = set [0];;

val xs : Set<int> = set [0]

> Gen.shuffle xs |> Gen.sample 1000 10;;
val it : seq<int> list =
  [seq [0]; seq [0]; seq [0]; seq [0]; seq [0]; seq [0]; seq [0]; seq [0];
   seq [0]; seq [0]]
```

Interestingly, the property still passes:

```F#
> let xs = set [0];;

val xs : Set<int> = set [0]

> Gen.shuffle xs |> Gen.sample 1000 10 |> List.exists ((<>) (Set.toSeq xs));;
val it : bool = true
```

Investigating further, so does a variation with `forall`:

```F#
> let xs = set [0];;

val xs : Set<int> = set [0]

> Gen.shuffle xs |> Gen.sample 1000 10 |> List.forall ((<>) (Set.toSeq xs));;
val it : bool = true
```

The reason, it seems, is that sequences don't have structural equality (neither would I expect them to):

```F#
> let xs = set [0];;

val xs : Set<int> = set [0]

> seq [0] = Set.toSeq xs;;
val it : bool = false
```